### PR TITLE
fixed overlay plots and stacked plots

### LINKED ISSUE #675: Plotting Vector's, Quaternions, and Booleans in MCT
Currently the built in plots and graphs in OpenMCT do not support graphing vectors, quaternions, or booleans, so plugins must be written to create this functionality.

Possiblities for each:

Booleans:
 - Simple conversion to 1 and 0 and then plot using build in numerical plots
 - pictorial representation (ex. turning on a light or changing the color of some box)
 
Vector: 
 - splitting the vectors into their components and then creating stacked plots for these.
 - overlay the three components on the same plot (like matplotting them in ptest does it)
 - having fields and plots to show magnitudes etc that are calculated on ground
 
Quaternions:
- same ideas as vectors but more Quaterniony lol

### PR DESCRIPTION
**Pull request by Duncan-McD**
# Fixes Overlay Plots in MCT

Fixes #675 pt2

### Summary of changes
- Made everything a domain object instead of having subfields
- this required refactoring both the json generator and a large portion of the telemetry.js file

If it applies, take a chance to describe offshoot work or TODOs that are a part of your PR.

### Testing

tested with multiple ptest cases and now both overlay plots and stacked plots work since everything is its own domain object:

Screenshot from 2021-05-08 07-28-43

### Constants

N/A

### Telemetry

N/A

### Documentation Evidence

I added some inline documentation to better understand the refactoring

### CODE REVIEW

**Comment by Duncan-McD** [comment]
Also just noticed I must have had a format on save on for telemetry.js - I apologize

File: MCT/server-files/telemetry.js [hunk 1]
```diff
@@ -220,141 +250,294 @@ Telemetry.prototype.getValue = async function (myUrl, i, f) {
...
+
+        //create z telempoint
+        var telempointz = { timestamp: timestamp, id: 'leader_z_' + id.substring(7) };
+        //set value
+        telempointz['value'] = getCoord(answer, 3)
+        //notify the realtime server and push the datapoint to the history server
+        this.notify(telempointz);
+        if(this.history['leader_z_' + id.substring(7)] == undefined){ //makes sure it is not undefined
+        this.history['leader_z_' + id.substring(7)] = []
+        }
```
Comment (shihaocao) [inline MCT/server-files/telemetry.js:468]: is there a way to abstract this a bit with a for loop + a list of like `[leader_x_, leader_y_, leader_z_]` or something, seems like a lot of repeated code

Comment (Duncan-McD) [reply MCT/server-files/telemetry.js:468]: yeah sure - will do that soon

Comment (Duncan-McD) [reply MCT/server-files/telemetry.js]: fixed this

**Review by shihaocao — APPROVED** [review]

**Pull request merged**